api/envoy/config/route/v3/route_components.proto

```diff
@@ -750,10 +750,54 @@ message RouteAction {
     ConnectConfig connect_config = 3;
   }
 
+  message ClusterSelector {
+    oneof selector {
+      option (validate.required) = true;
+
+      // Indicates the upstream cluster to which the request should be routed
+      // to.
+      string cluster = 1 [(validate.rules).string = {min_bytes: 1}];
+
+      // Envoy will determine the cluster to route to by reading the value of the
+      // HTTP header named by cluster_header from the request headers. If the
+      // header is not found or the referenced cluster does not exist, Envoy will
+      // return a 404 response.
+      //
+      // .. attention::
+      //
+      //   Internally, Envoy always uses the HTTP/2 *:authority* header to represent the HTTP/1
+      //   *Host* header. Thus, if attempting to match on *Host*, match on *:authority* instead.
+      string cluster_header = 2 [
+        (validate.rules).string = {min_bytes: 1 well_known_regex: HTTP_HEADER_NAME strict: false}
+      ];
+
+      // Multiple upstream clusters can be specified for a given route. The
```

Comment:
I think we need some worked examples on how fallback is going to happen with `WeightedCluster`, either here or in the routing architecture docs. Basically, it needs to explain that we keep rolling the dice at each successive fallback, rather than rolling the dice once with the options filtered down by cluster liveness.